data-pipeline-flink/dp-core/src/test/scala/org/sunbird/spec/BaseProcessFunTestSpec.scala

```diff
@@ -0,0 +1,78 @@
+        |""".stripMargin
+
+    try {
+      val config = ConfigFactory.load("test.conf");
+      val bsConfig = new BaseProcessTestConfig(config)
+
+      val kafkaConnector = new FlinkKafkaConnector(bsConfig)
+
+      implicit val env: StreamExecutionEnvironment = FlinkUtil.getExecutionContext(bsConfig)
+      val userDefinedConfig = EmbeddedKafkaConfig(kafkaPort = 9092, zooKeeperPort = 2181)
+
+      val mapStream: SingleOutputStreamOperator[util.Map[String, AnyRef]] =
+        env.addSource(kafkaConnector.kafkaMapSource(bsConfig.kafkaMapInputTopic), "telemetry-raw-events-consumer")
+          .rebalance().keyBy(key => key.get("partition").asInstanceOf[Integer])
+          .process(new TestMapStreamFunc(bsConfig)).name("TestFun")
+
+      val eventStream = env.addSource(kafkaConnector.kafkaEventSource[Event](bsConfig.kafkaEventInputTopic), "kafka-telemetry-denorm-consumer")
+        .rebalance().keyBy(key => key.getPartition)
+        .process(new TestEventStreamFunc(bsConfig)).name("DruidValidator")
+
+      mapStream.getSideOutput(bsConfig.mapOutPutTag)
+        .addSink(kafkaConnector.kafkaMapSink(bsConfig.kafkaMapOutPutTopic))
+        .name("kafka-telemetry-failed-events-producer")
+
+      eventStream.getSideOutput(bsConfig.eventOutPutTag)
+        .addSink(kafkaConnector.kafkaEventSink[Event](bsConfig.kafkaEventOutPutTopic))
+        .name("Event-Producer")
+
+      withRunningKafkaOnFoundPort(userDefinedConfig) { implicit actualConfig =>
+        createCustomTopic(bsConfig.kafkaMapInputTopic)
+        createCustomTopic(bsConfig.kafkaMapOutPutTopic)
+        createCustomTopic(bsConfig.kafkaEventInputTopic)
+        createCustomTopic(bsConfig.kafkaEventOutPutTopic)
+        createCustomTopic(bsConfig.kafkaMetricsOutPutTopic)
+        publishStringMessageToKafka(bsConfig.kafkaMapInputTopic, EVENT_WITH_MESSAGE_ID)
+        publishStringMessageToKafka(bsConfig.kafkaEventInputTopic, SHARE_EVENT)
+
+        Future {
+          env.execute("TestFlinkProcess Job")
+        }
+        println("Output from Map process func" + consumeFirstStringMessageFrom(bsConfig.kafkaMapOutPutTopic))
```

Comment:
Can you add some assertions over here...you may need to apply a retry loop over here and consume all the messages within the topic